### ISSUE TITLE
Bug: AlphaBlend target i686-pc-windows-msvc does not link Msimg32.lib

### ISSUE THREAD

**Issue by ganlvtech** [microsoft/windows-rs#1669]
### Which crate is this about?

windows

### Crate version

3

### Summary

[AlphaBlend](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/nf-wingdi-alphablend) with target `i686-pc-windows-msvc` does not link `Msimg32.lib`. It works very well with `x86_64-pc-windows-msvc` target.

I'm using `stable-x86_64-pc-windows-msvc` toolchain and `i686-pc-windows-msvc` target.

* `cargo build` ☺
* `cargo build --target=i686-pc-windows-msvc` ☹

I can make it compile by adding `link-arg` in `.cargo/config` manually.

```toml
[target.i686-pc-windows-msvc]
rustflags = ["-C", "link-arg=Msimg32.lib"]
```

### Toolchain version/configuration

Default host: x86_64-pc-windows-msvc
rustup home:  C:\Users\rust\.rustup

installed toolchains
--------------------

stable-x86_64-pc-windows-msvc (default)
nightly-x86_64-pc-windows-msvc

installed targets for active toolchain
--------------------------------------

i686-pc-windows-msvc
x86_64-pc-windows-msvc

active toolchain
----------------

stable-x86_64-pc-windows-msvc (default)
rustc 1.59.0 (9d1b2106e 2022-02-23)

### Reproducible example

```rust
use std::thread::sleep;
use std::time::Duration;

use windows::Win32::Foundation::HWND;
use windows::Win32::Graphics::Gdi::{AC_SRC_ALPHA, AC_SRC_OVER, AlphaBlend, BLENDFUNCTION, GetDC, ReleaseDC};

fn main() {
    unsafe {
        let hwnd = HWND(0);
        let hdc = GetDC(hwnd);
        for _ in 0..1000 {
            AlphaBlend(hdc, 100, 100, 100, 100, hdc, 0, 0, 100, 100, BLENDFUNCTION {
                BlendOp: AC_SRC_OVER as u8,
                BlendFlags: 0,
                SourceConstantAlpha: 128,
                AlphaFormat: AC_SRC_ALPHA as u8,
            });
            sleep(Duration::from_millis(16));
        }
        ReleaseDC(hwnd, hdc);
    }
}
```

### Crate manifest

```toml
[package]
name = "windows-rs-demo"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]

[dependencies.windows]
version = "0.35.0"
features = [
    "Win32_Foundation",
    "Win32_Graphics_Gdi",
]
```

### Expected behavior

Compiled

### Actual behavior

```plain
error: linking with `link.exe` failed: exit code: 1120
  |
  = note: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX64\\x86\\link.exe" "/NOLOGO" "/LARGEADDRESSAWARE" "/SAFESEH" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.windows_rs_demo.70fba944-cgu.0.rcgu.o" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.windows_rs_demo.70fba944-cgu.1.rcgu.o" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.windows_rs_demo.70fba944-cgu.10.rcgu.o" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.windows_rs_demo.70fba944-cgu.11.rcgu.o" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.windows_rs_demo.70fba944-cgu.12.rcgu.o" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.windows_rs_demo.70fba944-cgu.13.rcgu.o" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.windows_rs_demo.70fba944-cgu.14.rcgu.o" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.windows_rs_demo.70fba944-cgu.15.rcgu.o" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.windows_rs_demo.70fba944-cgu.2.rcgu.o" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.windows_rs_demo.70fba944-cgu.3.rcgu.o" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.windows_rs_demo.70fba944-cgu.4.rcgu.o" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.windows_rs_demo.70fba944-cgu.5.rcgu.o" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.windows_rs_demo.70fba944-cgu.6.rcgu.o" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.windows_rs_demo.70fba944-cgu.7.rcgu.o" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.windows_rs_demo.70fba944-cgu.8.rcgu.o" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.windows_rs_demo.70fba944-cgu.9.rcgu.o" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.3h12yg0ovr2qmnyc.rcgu.o" "/LIBPATH:C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps" "/LIBPATH:C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\debug\\deps" "/LIBPATH:C:\\Users\\rust\\.cargo\\registry\\src\\mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd\\windows_i686_msvc-0.35.0\\lib" "/LIBPATH:C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib" "C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\libwindows-7d5b200f61444baf.rlib" "C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libstd-8f8fecc910ffaf57.rlib" "C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libpanic_unwind-f1884aa05ba41556.rlib" "C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libstd_detect-bfb734f400da71e7.rlib" "C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\librustc_demangle-bdda46d8be98e0d0.rlib" "C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libhashbrown-52435b45e0de6e8f.rlib" "C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\librustc_std_workspace_alloc-a1234dd8fcb54345.rlib" "C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libunwind-908a8dd78ac3a16e.rlib" "C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libcfg_if-06ea2bb174518a09.rlib" "C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\liblibc-9bb42bbd818b6730.rlib" "C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\liballoc-14b20b39f7fe2525.rlib" "C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\librustc_std_workspace_core-8a3d877bba00ac17.rlib" "C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libcore-e840e28c14c9e14b.rlib" "C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib\\libcompiler_builtins-a2c0b883f1504c66.rlib" "windows.lib" "kernel32.lib" "ws2_32.lib" "bcrypt.lib" "advapi32.lib" "userenv.lib" "kernel32.lib" "msvcrt.lib" "/NXCOMPAT" "/LIBPATH:C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\i686-pc-windows-msvc\\lib" "/OUT:C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.exe" "/OPT:REF,NOICF" "/DEBUG" "/NATVIS:C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libcore.natvis" "/NATVIS:C:\\Users\\rust\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libstd.natvis"
  = note: error LNK2019: unresolved external symbol '__imp__AlphaBlend@44' referenced in function '__ZN7windows7Windows5Win328Graphics3Gdi10AlphaBlend17heb03e6e2f680ad36E'
  C:\\Users\\rust\\Rust\\windows-rs-demo\\target\\i686-pc-windows-msvc\\debug\\deps\\windows_rs_demo.exe : fatal error LNK1120: 1 unresolved externals
```

**Comment by kennykerr** [comment]
Thanks for reporting! Had a quick look and can confirm the issue.

It's looking for `__imp__AlphaBlend@44` which looks like the correct mangled name and should be included in the bundled `windows.lib`, but checking `windows.lib` it seems to have the wrong parameter size (56 instead of 44):

```
D:\alpha>dumpbin /exports d:\git\windows-rs\crates\targets\i686_msvc\lib\windows.lib | findstr /i _AlphaBlend
                  _AlphaBlend@56

D:\alpha>dumpbin /exports "C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22000.0\um\x86\msimg32.lib" | findstr /i _AlphaBlend
                  _AlphaBlend@44
```

**Comment by kennykerr** [comment]
Thanks again for reporting this issue. It has been fixed with #1674.

**Comment by kennykerr** [comment]
[0.36.0](https://github.com/microsoft/windows-rs/releases/tag/0.36.0) has now been reased and includes this fix.
